google/cloud/storage/retry_policy_test.cc

```diff
@@ -0,0 +1,25 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/retry_client.h"
+#include <gmock/gmock.h>
+
+using namespace storage;
+
+TEST(RetryPolicyTest, PermanentFailure) {
+  EXPECT_TRUE(StatusTraits::IsPermanentFailure(Status(404, "not-found")));
```

Comment:
Might be worth adding in a more exhaustive set of status codes... specifically, the ones we document that we return (neat, we even recommend retry strategies...) [1]: 400, 401, 403-405, 409-413, 416, 429, 500, 502, 503

[1] https://cloud.google.com/storage/docs/json_api/v1/status-codes

Reply:
Fixed.